stackit/internal/services/serverbackup/schedule/datasource.go

```diff
@@ -0,0 +1,240 @@
+	return &scheduleDataSource{}
+}
+
+// scheduleDataSource is the data source implementation.
+type scheduleDataSource struct {
+	client *serverbackup.APIClient
+}
+
+// Metadata returns the data source type name.
+func (r *scheduleDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_serverbackup_schedule"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *scheduleDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerData, ok := req.ProviderData.(core.ProviderData)
+	if !ok {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Expected configure type stackit.ProviderData, got %T", req.ProviderData))
+		return
+	}
+
+	var apiClient *serverbackup.APIClient
+	var err error
+	if providerData.ServerBackupCustomEndpoint != "" {
+		apiClient, err = serverbackup.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.ServerBackupCustomEndpoint),
+		)
+	} else {
+		apiClient, err = serverbackup.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the data source configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Serverbackup schedule client configured")
+}
+
+// Schema defines the schema for the data source.
+func (r *scheduleDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Serverbackup schedule data source schema.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Terraform's internal data source identifier. It is structured as \"`project_id`,`server_id`\".",
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT Project ID (UUID) to which the server is associated.",
+				Required:    true,
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"server_id": schema.StringAttribute{
+				Description: "Server ID (UUID) to which the backup schedule is associated.",
+				Required:    true,
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"items": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"backup_schedule_id": schema.Int64Attribute{
+							Computed: true,
+						},
```

Comment:
We have never had a case where the datasource is so different than the resource. So I'm not sure I understand the use case, is there a reason why this datasource does not take the `backup_schedule_id` as a required field and reads the data of a single backup schedule?

Reply:
Yes. So, I will do as you suggest (`backup_schedule_id` as a required field, populate only 1 schedule) to conform with the rest of the project.

---

In the future, probably the project will need to think of how to filter a list of backup schedules. Like, with a datasource - adding a bunch of `ids` into terraform. There's probably no need to d this as part of this PR. 

For completeness, to explain what I mean: it seems at other terraform providers over the internet, this happens in the following ways (maybe more, this is what I found):

1. a `filter` attribute with an optional `id` attribute (like: [aws_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/instance), etc.)  or
2. adding a plural version of the datasource. Like, having `stackit_server_backup_schedule` and a `stackit_server_backup_schedules` datasource. 
Examples:
[oci_core_subnets](https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/core_subnets) <-> [oci_core_subnet](https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/core_subnet) 
[aws_vpcs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/vpcs) <-> [aws_vpc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/vpc)

With the required `id` it seems like you have chosen the second way - it's ok with me. If you haven't decided on it yet, this can be added at a later PR maybe, etc.

Thanks for noticing it in the comment, too.

Reply:
> ok - I will do as you suggest (backup_schedule_id as a required field, populate only 1 schedule) to conform with the rest of the project.

Alright, thank you

> a filter attribute with an optional id attribute (like: [aws_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/instance), etc.) or

I didn't know this approach, but it also says that `If anything other than a single match is returned by the search, Terraform will fail. Ensure that your search is specific enough to return a single Instance ID only.`

> adding a plural version of the datasource. Like, having stackit_server_backup_schedule and a stackit_server_backup_schedules datasource.

So I would prefer this one if we want to allow getting data of multiple backups 👍

Reply:
You have already implemented it, right? So you could even leave it and change the name to be plural, and implement the singular one. What do you think @aeter